### PR TITLE
Saycode Rework

### PR DESCRIPTION
**Pull request by N3X15**
WIP, have some major bugs to fix still.
# CHANGES
- Speech now uses the new /datum/speech for shuttling around speech data.
- Formatting is no longer injected as HTML into the message.  HTML will be escaped and presented as normal text.  Formatting is kept separate from the message.
- Speech is no longer rendered, encoded, unencoded, filtered, and re-encoded.  Characters are left raw and unencoded until they reach someone's ears.
- Saycode logic flow documented. See docs/.
- Mutations and diseases can affect more of the message, including adding and removing classes without parsing HTML.

tl;dr eliminates  a lot of the `Guy says, "<SPAN!!!"` stuff.

### CODE REVIEW

**Comment by N3X15** [comment]
I'm currently dealing with the GC nuking the datums early.  However, I'm having a hard time getting work done because some fucking recruiter keeps calling me.

**Comment by Probe1** [comment]
Post testing that needs done as needed and I will test.

**Comment by mrpain666** [comment]
Clowns talking in Comic Sans when?

**Comment by N3X15** [comment]
Okay, I locally tested:
 * say verb
  * Local speech
  * Radios
  * Intercoms
 * whisper verb
 * Vending machines (mildly buggy when intercom in vicinity)

Should be good to go.

**Comment by d3athrow** [comment]
if the gc is nuking the datums give them a tag.

**Comment by N3X15** [comment]
No, it's fixed.

**Comment by N3X15** [comment]
```
<N3X15> merge saycode rewrite y/n
<dylanstrategie> Y
<Zth> y
<dylanstrategie> This way we can get it tested and hotfixed
<dylanstrategie> Not the first time we merge a massive rework that isn't fully functional either way
```
